src/UI/NavigationContainer.elm

```diff
@@ -434,13 +437,37 @@ Clicking on the black layer also activates the closing message.
 -}
 dialog : String -> msg -> Element msg -> Dialog msg
 dialog title onClose body =
-    Dialog
+    Dialog1
         { title = title
         , close = onClose
         , body = body
         }
 
 
+{-|
+
+    `Nav.dialogV2` constructs a [`Nav.Dialog`](#Dialog) from a title and icon.
+    This variant of the dialog does not require you to specify body at the time
+    of creation and you can specify it alongside dialog buttons as an option
+    like this:
+
+    Nav.dialog2 "An example dialog" Icon.warning
+        |> withBody (Element.text "Dialog body")
+        |> withButtons [submitButton, cancleButton]
+
+-}
+dialogV2 : String -> Icon -> Dialog msg
```

Comment:
I think it makes more sense to receive `UI.V2.Dialog.Dialog` instead of `String -> Icon`, this way the user will be able to set the dialog the way they want

Reply:
If we make it receive `UI.V2.Dialog.Dialog`, then having the helpers like `UI.V2.withBody` and `UI.V2.withButtons` are needless I think. Perhaps you meant that `dialogV2` should receive `UI.V2.Properties`?

Reply:
@jouderianjr notice that the dialogV2 is not some `UI.V2.Dialog -> Nav.Container -> Nav.Container`, it's the constructor that should actually been in `UI.V2.Dialog.dialog`.